### ISSUE TITLE
Allow for the embed option on the serializer to do nothing 👍

### ISSUE THREAD

**Issue by cah-briangantzler** [miragejs/ember-cli-mirage#1941]
currently the embed option allows for true (embed the records) or false (sideload the records).   What I would like is half the sideload. IE, serialize the record IDs but dont include the side loaded records.   For me the relation records are already loaded, or will be requested in their own request.     

I also need root: true.   So what I am currently doing is root:false, embed: false and then in the serializer code I do the root myself and throw away the side loaded models (wasting the time serializing them in the first place)

Proposal: Alter the embed option to be

- sideload - (default) does what the current false does
- true - does what the current true does
- false - serializes the ids, do nothing with the related record.

You would change the error message to not allow root: false and embed: sideload

I think this would also be the most backward compatible in that Im hoping people did not put embed: false in their code as it would be the default.

Alternatively you could create a new option, sideload: true (default).   I think that would be confusing as embed:true and sideload:true would produce both.

If this is an acceptable request, and you decide how you would like this implemented, would be happy to attempt a PR.

**Comment by cah-briangantzler** [comment]
I am not sure if this might belong in mirageJS as this might apply to other environments.

**Comment by samselikoff** [comment]
Believe you're looking for https://miragejs.com/api/classes/serializer/#serialize-ids

**Comment by cah-briangantzler** [comment]
I did read this and did not draw the same conclusion. 

Are you saying for the serialzier 
```
{
     root: true, 
     embed:false, 
     serializeIds: always
}
 ```
 will get me JSON output of
```
{
     attr1: "data"
     fk1: "12"
}
```

**Comment by samselikoff** [comment]
Yes exactly, it should!

**Comment by cah-briangantzler** [comment]
Ill give it a shot, but I found this https://stackoverflow.com/questions/60980497/is-there-a-way-to-show-related-model-ids-without-sideloading-or-embedding-data and Im thinking it wont. I have to write my own root code in my application serializer, which means Im serializing all that data just to throw it away, was hoping to save that time by my proposal. 

Worse though, I just ran into an API where one of the properties is embedded and the rest are not. The embed option is an all or nothing, but ED provides a property by property. (im thinking non-ember people would have this problem as well. Backends can be very inconsistent).

from an ember data serializer 

```
 attrs: {
    name: {key: "facilityName"},
    reorderPoint: {key: "flagReorderPoint"},
    lotAndExpirationView: {key: "flagLotAndExpirationView"},
    ndcSubItemsView: {key: "flagNdcSubItemsView"},
    network: {key: "networkId"},  // this is the default of 'ids'
    account: {deserialize: "records", serialize: "records"},
  }
```
plus it allows renaming properties (FK and attr) pretty concisely. Would be nice instead of writing a custom keyForAttribute and keyForRelationship (and possible a couple others). 

So at this point, Ill see what I can do about writing a brand new serializer that might be able to be merged in mirageJs proper.

once you respond I think we can close this issue (it just raises so many more LOL)
